app/models/post_action_type.rb

```diff
@@ -46,7 +51,11 @@ def overridden_by_plugin_or_skipped_db?
     end
 
     def all_flags
-      Flag.unscoped.order(:position).all
+      current_db = RailsMultisite::ConnectionManagement.current_db
+      @all_flags_ordered ||= {}
+      @all_flags_ordered[current_db] ||= begin
+        Flag.unscoped.order(:position).all
```

Comment:
A question for @lis2 when he is back but is there a limit on the number of flags we allow per site? We're caching every possible flag in memory per process and this worries me.

Reply:
I don’t think we have a limit but we should, this is not really a thing where admins should be adding more than twenty or so custom flags

Reply:
There are 3 methods which calls `ActiveRecord#pluck` on `all_flags` and I see 2 of those methods being used in serializers. 

https://github.com/discourse/discourse/blob/11369018b6841b7afa1a564858cbefb710b05be9/app/models/post_action_type.rb#L117-L127

Calling `ActiveRecord#pluck` will always trigger a DB query so memoizing here may not be sufficient.

Reply:
We could switch these to `.map` instead since the data is already loaded